### PR TITLE
Added blog post about bingo.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

As usual, up to date preview: https://deploy-preview-16--bwplotka.netlify.app/2020/bingo/

### CODE REVIEW

File: web/content/posts/bingo.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+- posts
+title: Require a Go Tool for Your Project? That's a Bingo! 
+weight: 1
+categories:
+- go
+featuredImage: "/images/blog/bingo/demo.gif"
+---
+_See bigger version of the above demo as .gif [here](https://raw.githubusercontent.com/bwplotka/bingo/master/examples/bingo-demo.gif)_
+
+In this blog post I would like to introduce [`bingo`](https://github.com/bwplotka/bingo). Simple and efficient CLI I wrote for managing versioning of Go binaries that are required for your project development.
```
Comment (soniasingla) [inline web/content/posts/bingo.md:17]: In this blog post I would like to introduce [`bingo`](https://github.com/bwplotka/bingo), simple and efficient CLI (Command Line Tool), I wrote for managing versioning of Go binaries that are required for your project development.

File: web/content/posts/bingo.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+**TL;DR: `bingo`is built on top of native [Go Modules dependency management](https://github.com/golang/go/wiki/Modules) and in my option it
+ is a must-have for all repositories requiring dev tools written in Go. It's also already integrated into production grade projects like [Thanos](https://thanos.io). Check it out and contribute! 🤗**
```
Comment (soniasingla) [inline web/content/posts/bingo.md:20]: **TL;DR: `bingo`is built on top of native [Go Modules dependency management](https://github.com/golang/go/wiki/Modules) and in my option it
 is a must-have tool for all repositories which requires dev tools written in Go. It's also already integrated into production grade projects like [Thanos](https://thanos.io). Check it out and contribute! 🤗**

File: web/content/posts/bingo.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+# Automate all the things!
+
+![This is NOT  what you want to do in your project...](/images/blog/bingo/automation.gif)
+
+> Automation is **amazing**. 
+
+Especially nowadays, all open or closed projects use at least bunch of tools to automate some tasks. For example in Go community
+you can see following popular tools enabling automation:
```
Comment (soniasingla) [inline web/content/posts/bingo.md:29]: Especially nowadays, all projects whether open or closed use at least bunch of tools to automate some tasks. For example, in Go community,
you can see following popular tools enabling automation:

File: web/content/posts/bingo.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+In this blog post I would like to introduce [`bingo`](https://github.com/bwplotka/bingo). Simple and efficient CLI I wrote for managing versioning of Go binaries that are required for your project development.
+
+**TL;DR: `bingo`is built on top of native [Go Modules dependency management](https://github.com/golang/go/wiki/Modules) and in my option it
+ is a must-have for all repositories requiring dev tools written in Go. It's also already integrated into production grade projects like [Thanos](https://thanos.io). Check it out and contribute! 🤗**
+
+# Automate all the things!
+
+![This is NOT  what you want to do in your project...](/images/blog/bingo/automation.gif)
+
+> Automation is **amazing**.
```
Comment (soniasingla) [inline web/content/posts/bingo.md:26]: We can align this highlight in center.

File: web/content/posts/bingo.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+* Auto formatters: `go fmt`, [`goimports`](https://godoc.org/golang.org/x/tools/cmd/goimports)
+* Static analysis: `go vet`, [`lint`](https://github.com/golang/lint), [`golangci-lint`](https://github.com/golangci/golangci-lint), [`faillint`](https://github.com/fatih/faillint)
+* Documentation Generators: [`embedmd`](https://github.com/campoy/embedmd), [docs auto generators](https://github.com/thanos-io/thanos/tree/master/scripts/cfggen)
+* Build tools: [`goreleaser`](https://github.com/goreleaser/goreleaser), [`promu`](https://github.com/prometheus/promu)
+* Other projects (!) you want to run integrations tests against.
+* ...and millions of other tools: [website builder](https://gohugo.io/), [link check](https://github.com/raviqqe/liche), [copyright checker](https://github.com/thanos-io/thanos/tree/master/scripts/copyright), protobuf + gRPC generation, jsonnet generators and bundlers, [YAML tools](https://github.com/brancz/gojsontoyaml), etc... 
+
+I could list even more, and I am so amazed with the amount of things we have **for free**! They help maintaining very good quality of the
+project, and saves an enormous amount of time.
```
Comment (soniasingla) [inline web/content/posts/bingo.md:41]: I could list even more, and I am so amazed with the amount of things we have **for free**! They help in maintaining very good quality of the
project, and saves an enormous amount of time.

**Comment by soniasingla** [comment]
I am not able to directly edit the lines in respective file :(

File: web/content/posts/bingo.md (hunk 5 above)
Comment (bwplotka) [reply web/content/posts/bingo.md]: repeating `amount` ;p

**Comment by bwplotka** [comment]
Thanks @soniasingla for review :hugs: BTW, you can use the `suggestion` feature to give comments like this much easier (: Did you try that?

**Comment by bwplotka** [comment]
BTW @soniasingla I added the rest of the content if you want to take a look.

File: web/content/posts/bingo.md [hunk 6]
```diff
@@ -0,0 +1,399 @@
+---
+authors:
+- name: "Bartek Płotka"
+date: 2020-06-02
+linktitle: Require a Go Tool for Your Project? That's a Bingo! 
+type:
+- post 
+- posts
+title: Require a Go Tool for Your Project? That's a Bingo!
```
Comment (metalmatze) [inline web/content/posts/bingo.md:9]: I think "Requiring a Go Tool for your project?" would make more sense. That @squat  probably knows best :smile:

File: web/content/posts/bingo.md [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+- posts
+title: Require a Go Tool for Your Project? That's a Bingo! 
+weight: 1
+categories:
+- go
+featuredImage: "/images/blog/bingo/demo.gif"
+---
+_See bigger version of the above demo as .gif [here](https://raw.githubusercontent.com/bwplotka/bingo/master/examples/bingo-demo.gif)_
+
+In this blog post, I would like to introduce [`bingo`](https://github.com/bwplotka/bingo), simple and efficient CLI (Command Line Tool),
```
Comment (metalmatze) [inline web/content/posts/bingo.md:17]:
```suggestion
In this blog post, I would like to introduce [`bingo`](https://github.com/bwplotka/bingo), a simple and efficient CLI (Command Line Tool),
```

Comment (squat) [reply web/content/posts/bingo.md:17]: not to be too strict, but Bingo is not really a CLI; it _has_ a CLI. Specifically, it is a CLI tool

File: web/content/posts/bingo.md [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+I wrote for managing versions of Go binaries that are required for your project development.
+
+**TL;DR: `bingo` is built on top of the native [Go Modules dependency management](https://github.com/golang/go/wiki/Modules) and in my opinion,
+it is a must-have tool for all repositories requiring dev tools written in Go. It's also already integrated into production-grade projects like [Thanos](https://thanos.io).
```
Comment (metalmatze) [inline web/content/posts/bingo.md:21]:
```suggestion
it solves the hard problem of requiring dev tools written Go when you might have a different version of the same dependency. It already improves our integration of tools in [Thanos](https://thanos.io).
```

File: web/content/posts/bingo.md (hunk 6 above)
Comment (bwplotka) [reply web/content/posts/bingo.md:9]: :+1: LUCAS HELP @squat ;p

Comment (bwplotka) [reply web/content/posts/bingo.md:9]: Feel free @squat to review, will address comments later PR

File: web/content/posts/bingo.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+* It is an official tool, which means finally a single standard. They are also built-in other Go tools. 
+
+But let me say this loud and clear: **It's also far from being perfect.** Consider the following reasons:
+
+* It assumes everyone uses **semantic versioning** properly. 
+
+> Halo Go Team, can I have your attention for a second? 🤗 
+
+**99.9% of Go projects do NOT use semantic versioning properly and NEVER will be!** 
+
```
Comment (povilasv) [inline web/content/posts/bingo.md:198]: Never will be => Never will

File: web/content/posts/bingo.md [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+It's because maintainers either have no time, don't care or they simply semantically version their APIs or part of packages only. See [detailed 
+discussion on this matter in Prometheus (30k stars' project) mailing list.](https://groups.google.com/forum/#!searchin/prometheus-developers/Go$20Modules%7Csort:date/prometheus-developers/F1Vp0rLk3TQ/TyF2WxlkBgAJ)
+
+This leads to most of the problems users experience. You can't escape from a huge amount of `replace` hacks.
+Plus, instead of making it easier for such use cases, Go blames others. ):
+
+This is also why `bingo` was needed and why I built it.
+
+* We build and import packages, but version modules. If you add an overhead of maintaining modules and releasing it (see the above issue),
+it's clear that maintaining multiple modules is not a good answer. That's why [Duco's](Helcaraxan) amazing [`modularise`](https://github.com/modularise/modularise)
```
Comment (povilasv) [inline web/content/posts/bingo.md:208]: [Duco's](Helcaraxan) links to broken page => https://www.bwplotka.dev/2020/bingo/Helcaraxan

**Pull request merged**